Ginger/GingerCoreCommon/VariablesLib/VariableBase.cs

```diff
@@ -212,7 +212,8 @@ public string Formula
         public abstract string GetFormula();
         public abstract string VariableType { get; }
         public abstract void ResetValue();
-        public abstract void GenerateAutoValue();
+        private string errorMsg;
+        public abstract void GenerateAutoValue(ref string errorMsg);
```

Comment:
i'm thinking why not also to replace 'void' with bool to indicate if operation succeed or not, let talk on it.